aidb/engine/approx_select_engine.py

```diff
@@ -0,0 +1,211 @@
+import logging
+import math
+import multiprocessing as mp
+import numpy as np
+import pandas as pd
+from sqlalchemy.sql import text
+from typing import List
+
+from aidb.engine.tasti_engine import TastiEngine
+from aidb.query.query import Query
+from aidb.utils.constants import VECTOR_ID_COLUMN
+
+PROXY_SCORE = 'proxy_score'
+MASS = 'mass'
+WEIGHT = 'weight'
+BUDGET = 2000
+
+class ApproxSelectEngine(TastiEngine):
+  # TODO: design a better algorithm, this function is same as the function in Limit Engine
+  def score_fn(self, score_for_all_df: pd.DataFrame, score_connected: List[List[str]]) -> pd.Series:
```

Comment:
I think this is a big reason why the score is bad. Can you actually fix it in this PR?

Reply:
Although I'm not quite sure what's going on here

Reply:
This function is used to get a final score for multiple predicates in WHERE clause. For single predicate, it doesn't matter. I think one reason is that in previous design, if one blob has several derived rows, some rows may not satisfy predicate, and I choose a mean score as that blob's proxy score, this result the average proxy score is low. In this case, should I set the blob's proxy score as 1 or as mean score?

Reply:
For counts, what's wrong with setting it to the estimated count? I think for derived rows, we want to estimate the count and do proportional importance sampling or something actually (or control variates)

Reply:
Can you raise an issue if you aren't going to resolve this now @ttt-77 ?